File: Binary Search/BS on 1D Arrays/Search_insert_position.java
public class Search_insert_position{
    public static void main(String[] args) {
        int [] arr = {1, 3, 5, 6};
        int x = 4;

        int low = 0, high = arr.length - 1;
        int ans = arr.length;

        while (low <= high) { 
            int mid = (low + high)/2;

            if (arr[mid] >= x) {
                ans = mid;
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        System.out.println("The position to insert " + x + " is: " + ans);
    }
}

File: Binary Search/BS on 1D Arrays/Python/Search_insert_position.py
def search_insert_position(nums, target):
    n = len(nums)

    low = 0 
    high = n - 1
    ans = n

    while low <= high:
        mid = int((low + high)/2)

        if nums[mid] >= target:
            ans = mid
            high = mid -1
        else:
            low = mid + 1
        
    return ans



nums = [1, 3, 5, 6]
target = 2
print(search_insert_position(nums, target))